src/nccl_ofi_ep_addr_list.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
+
+#include "config.h"
+
+#include "nccl_ofi.h"
+
+#include "nccl_ofi_ep_addr_list.h"
+#include "nccl_ofi_pthread.h"
+
+#include "contrib/uthash.h"
+#include "contrib/utlist.h"
+
+/**
+ * A Libfabric address, stored in a form hashable by uthash
+ */
+typedef struct {
+	char addr[MAX_EP_ADDR];
+	UT_hash_handle hh;
+} hashed_addr_t;
+
+/**
+ * A struct storing a pair of (ep, addr_set)
+ */
+typedef struct {
+	nccl_net_ofi_ep_t *ep;
+	/* Hash set of addresses the endpoint is connected to */
+	hashed_addr_t *addr_set;
+} pair_ep_addr_set_t;
```

Comment:
what's the difference between a `pair_ep_addr_set_t` and a `ep_pair_list_elem_t`?  Why are the names formatted differently?  Why are there two structures at all?

Reply:
Good point, these can be combined into the same structure. Will do.